crates/docs/src/validation.rs

```diff
@@ -1,15 +1,51 @@
 use regex::Regex;
 use std::{
+    collections::HashMap,
     env, fs, io,
     path::{Path, PathBuf},
 };
+use walkdir::WalkDir;
 
 const EXTENSION: Option<&str> = Some("md");
 
+#[derive(Clone, Debug)]
+pub struct SnippetType(String);
+
+impl SnippetType {
+    #[must_use]
+    pub fn forge() -> Self {
+        SnippetType("snforge".to_string())
+    }
+
+    #[must_use]
+    pub fn sncast() -> Self {
+        SnippetType("sncast".to_string())
+    }
+
+    #[must_use]
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+
+    #[must_use]
+    pub fn get_re(&self) -> Regex {
+        let escaped_command = regex::escape(self.as_str());
+        let pattern = format!(
+            r"(?ms)^(?:<!--\s*(.*?)\s*-->\n)?```shell\n\$ ({escaped_command} .+?)\n```(?:\s*<details>\n<summary>Output:<\/summary>\n\n```shell\n([\s\S]+?)\n```[\s]*<\/details>)?"
```

Comment:
Could you add a short comment explanation of this regex?

Reply:
Sure, added.